Samples/DynamicDependencies/DirectX/direct3d9ex/BackgroundThread.cpp

```diff
@@ -0,0 +1,380 @@
+﻿//--------------------------------------------------------------------------------------
+// File: BackgroundThread.cpp
+//
+// Sample showing how to use D3D9Ex advanced features.
+//
+// Copyright(c) Microsoft Corporation.
+// Licensed under the MIT License.
+//--------------------------------------------------------------------------------------
+#include <windows.h>
+#include <d3d9.h>
+#include <d3dx9.h>
+
+#define SAFE_RELEASE(p) { if(p) { (p)->Release(); (p) = NULL; } }
```

Comment:
ugh! use RAII types, like wil::com_ptr instead.

Reply:
I'm taking an existing (old) sample from win7 era and showing the minimal steps to update it with the WinAppSDK and use the DirectX MSIX framework package that's in the store.

Is there a pointer to a better version of this sample that already has these changes (and still uses apis that require the older DirectX9 runtime?